### PR TITLE
Move coverage to Jenkins from Travis

### PR DESCRIPTION
**Pull request by mbutrovich**
We're starting to time out coverage builds on Travis after merging #398. This attempts to move the coverage build to Jenkins.

### CODE REVIEW

**Comment by mbutrovich** [comment]
Doesn't seem to upload the report to codecov successfully from Jenkins.

File: Jenkinsfile [hunk 1]
```diff
@@ -44,6 +44,36 @@ pipeline {
...
+                        sh 'cd build && lcov --remove coverage.info \'/usr/*\' --output-file coverage.info'
+                        sh 'cd build && lcov --remove coverage.info \'*/build/*\' --output-file coverage.info'
+                        sh 'cd build && lcov --remove coverage.info \'*/third_party/*\' --output-file coverage.info'
+                        sh 'cd build && lcov --remove coverage.info \'*/benchmark/*\' --output-file coverage.info'
+                        sh 'cd build && lcov --remove coverage.info \'*/test/*\' --output-file coverage.info'
+                        sh 'cd build && lcov --remove coverage.info \'*/src/main/*\' --output-file coverage.info'
+                        sh 'cd build && lcov --list coverage.info'
+                        sh 'cd build && curl -s https://codecov.io/bash > ./codecov.sh'
+                        sh 'cd build && chmod a+x ./codecov.sh'
+                        sh 'cd build && /bin/bash ./codecov.sh -X gcov'
```
Comment (crd477) [inline Jenkinsfile]: Try changing this to:
`sh 'cd build && /bin/bash ./codecov.sh -X gcov -U "--proxy http://proxy.pdl.cmu.edu:3128/"'`

Comment (crd477) [reply Jenkinsfile]: Before you push any changes, I've got a test build (http://jenkins.db.cs.cmu.edu:8080/blue/organizations/jenkins/terrier/detail/PR-551/18/pipeline) in progress now that the PR is reopened.  If that works, then you could push the change I suggested.

**Comment by crd477** [comment]
Sorry, I remembered entirely wrong.  Those workers are setup for NAT so the proxy it not an issue at all.  Disregard everything I said about the proxy.

Jenkins requires some authentication token for the codecov site that is probably stored in whatever Travis account is used for this project.  If you can get access to it, send it to me privately and I'll put it into the Jenkins server.

**Pull request merged**